cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package io.strimzi.operator.cluster.operator.resource;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.zjsonpatch.JsonDiff;
+import io.strimzi.operator.cluster.model.KafkaConfiguration;
+import io.strimzi.operator.cluster.model.KafkaVersion;
+import io.strimzi.operator.cluster.model.ModelUtils;
+import io.strimzi.operator.common.operator.resource.AbstractResourceDiff;
+import org.apache.kafka.clients.admin.AlterConfigOp;
+import org.apache.kafka.clients.admin.Config;
+import org.apache.kafka.clients.admin.ConfigEntry;
+import org.apache.kafka.common.config.ConfigResource;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static io.fabric8.kubernetes.client.internal.PatchUtils.patchMapper;
+
+/**
+ * Computes a diff between the current config (supplied as a Map ConfigResource) and the desired config (supplied as a ConfigMap).
+ * An algorithm:
+ *  1. Create map from supplied desired ConfigMap
+ *  2. Fill placeholders (e.g. ${BROKER_ID}) in desired map
+ *  3a. Loop over all entries. If the entry is in IGNORABLE_PROPERTIES or entry.value from desired is equal to entry.value from current, do nothing
+ *      else add it to the diff
+ *  3b. If entry was removed from desired, add it to the diff with null value.
+ *  3c. If custom entry was removed, delete property
+ *
+ */
+public class KafkaBrokerConfigurationDiff extends AbstractResourceDiff {
+
+    private static final Logger log = LogManager.getLogger(KafkaBrokerConfigurationDiff.class);
+    private final Map<ConfigResource, Config> current;
+    private Collection<ConfigEntry> currentEntries;
+    private ConfigMap desired;
+    private KafkaConfiguration diff;
+    private KafkaVersion kafkaVersion;
+    private int brokerId;
+    Map<ConfigResource, Collection<AlterConfigOp>> updated = new HashMap<>();
+
+    /**
+     * These options are skipped because they contain placeholders
+     * 909[1-4] is for skipping all (internal, plain, secured, external) listeners properties
+     */
+    public static final Pattern IGNORABLE_PROPERTIES = Pattern.compile(
+            "^(broker\\.id"
+            + "|.*-909[1-4].ssl.keystore.location"
```

Comment:
All others seem to use `\\.`, just this one uses simply `.` ... is it needed or not?